src/dmd/todt.d

```diff
@@ -456,6 +456,14 @@ extern (C++) void Expression_toDt(Expression e, ref DtBuilder dtb)
         }
     }
 
+    /* https://issues.dlang.org/show_bug.cgi?id=12652
+       Non-constant hash initializers should have a special-case diagnostic
+     */
+    void visitAssocArrayLiteral(AssocArrayLiteralExp e)
+    {
+        e.error("Associative array literals currently cannot be initialized globally, in structs or in classes; try using 'enum' or a module constructor instead");
```

Comment:
This line is rather long, the sugession should be a second line using `errorSuplemental`.

Reply:
The error message should rather mention that it's impossible to initialize AA at compile time instead of runtime (the difference between [1],[2] and [3],[4]).

(I assume that this distinction should be understandable for non-compiler developers but other suggestions are welcome]